packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -49,33 +49,26 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         return name.replace("@", "").replace("/", "-");
     }
 
+    /**
+     * Take an unknown error object and add the appropriate info from it to the event
+     * NOTE - All properties will be copied over from the error object, except those specifically
+     * marked as pii on an instance of type LoggingError. Take care to sanitize the argument before passing in.
+     * @param event - Event being logged
+     * @param error - Error to extract info from
+     * @param fetchStack - Whether to fetch the current callstack if error.stack is undefined
+     */
     public static prepareErrorObject(event: ITelemetryBaseEvent, error: any, fetchStack: boolean) {
         if (error === null || typeof error !== "object") {
             event.error = error;
         } else {
-            // WARNING: Exceptions can contain PII!
-            // For example, XHR will throw object derived from Error that contains config information
-            // for failed request, including all the headers, and thus - user tokens!
-            // Extract only call stack, message, and couple network-related properties form error object
-
-            const errorAsObject = error as {
-                stack?: string;
-                message?: string;
-            };
-
+            const errorAsObject = error as Partial<LoggingError>;
             event.stack = errorAsObject.stack;
             event.error = errorAsObject.message;
 
-            // Error message can contain PII information.
-            // If we know for sure it does, we have to not log it.
-            if (error.containsPII) {
-                event.error = "Error message was removed as it contained PII";
-            } else if (error.getTelemetryProperties) {
-                const telemetryProps: ITelemetryProperties = error.getTelemetryProperties();
-                for (const key of Object.keys(telemetryProps)) {
-                    if (event[key] === undefined) {
-                        event[key] = telemetryProps[key];
-                    }
+            const telemetryProps = errorAsObject.getTelemetryProperties?.() ?? {};
```

Comment:
how do I get component path for internal builds? It has been invaluable in solving problems to get to it (via having it on the stack :))

Reply:
Haha you've been spying on it in the `Data_stack` column?  Hmmm I can try to do one better here and include some classification and hard-code package name as allowed.  I was trying to make a non-controversial step forward but I guess I need to preserve current behavior in some sense...